app/src/main/java/com/example/amandahinchman/avocadofacts/MainActivity.kt

```diff
@@ -1,42 +1,72 @@
 package com.example.amandahinchman.avocadofacts
 
+import android.os.AsyncTask
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
 import android.widget.Button
 import android.widget.TextView
+import java.lang.ref.WeakReference
 import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
     // Declare our View Variables
     private lateinit var factTextView: TextView
 
-    private val facts = arrayOf("fact 1",
-            "fact 2",
-            "fact 3",
-            "fact 4",
-            "fact 5",
-            "fact 6",
-            "fact 7",
-            "fact 8",
-            "fact 9",
-            "fact 10")
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         // Assign the Views from the layout
         factTextView = findViewById(R.id.avocadoFact)
 
-        var moarButton = findViewById<Button>(R.id.more)
+        val moarButton = findViewById<Button>(R.id.more)
         moarButton.setOnClickListener {
-            val randomGen = Random()
-            val randomNumber = randomGen.nextInt(10)
-            factTextView.text = facts.get(randomNumber)
+            NewAvocadoFactTask(this).execute()
         }
 
+    }
+
+    fun updateAvacadoFact(fact:String){
+        factTextView.text = fact
+    }
+
+    /**
+     * progress : Void
+     * params : Void
+     * result : String
+     * This is depicted the AsyncTask signature
+     */
```

Comment:
I'm a little unclear about this comment... OH Okay I see it now. Double check me to make sure the order is not mixed up, but I think this is what it ought to be?

Maybe modify to be like -

```
/**
* AsyncTask<Params, Progress, Result> signature interface: 
* - params: the type of the parameters sent to the task upon
*             execution
* - progress: the type of the progress units published during the
*             background computation
* - result: the type of the result of the background computation
*/
```